### PR TITLE
Clean up flake.nix

### PR DESCRIPTION
**Pull request by houstdav000**
Clean up the flake.nix file with some functions, change the outputs to use the new "default" output types, clean up corresponding flake template, and add a flake-template check.

Updated the flake.lock file as well to update ghc and have all checks pass.

Signed-off-by: David Houston <houstdav000@gmail.com>

### CODE REVIEW

**Comment by jtojnar** [comment]
Thanks. Though it would be better to have one commit for each individual change – the current patch is very hard to review.

File: flake.nix [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
+
+          checks.template-simple =
+            let
+              flake = import (self.templates.simple.path + /flake.nix);
+              flakeSelf = flake // flake.outputs;
+            in
+            (flake.outputs { inherit nixpkgs; self = flakeSelf; napalm = self.outputs; }).packages.${system}.hello-world;
+        }) // {
+
+      overlays.default = final: _: nixpkgs.lib.filterAttrs (_: nixpkgs) {
```
Comment (jtojnar) [inline flake.nix:32]: The filter predicate does not look right.

File: flake.nix [hunk 2]
```diff
@@ -1,45 +1,56 @@
...
+            pkgs = nixpkgs.legacyPackages.${system};
+          };
         in
         {
-          packages = {
-            inherit (pkgs.napalm)
-              hello-world hello-world-deps netlify-cli deckdeckgo-starter
-              bitwarden-cli napalm-registry
-              ;
+          packages = (nixpkgs.lib.filterAttrs (_: nixpkgs.lib.isDerivation) napalm) // {
```
Comment (jtojnar) [inline flake.nix:16]: I think I prefer the explicit package lists – that way it is clear what is exported and devshell is not included among packages.

**Comment by houstdav000** [comment]
I'll split up the major commits, though some are dependent on one-another.

**Comment by jtojnar** [comment]
Thanks. It should be sufficient to create a single PR for all the commits if they are related. The only requirement is that the commits themselves are atomic so that their diffs can be reviewed individually.

**Pull request closed without merging**